src/fty-msgbus-cli.cc

```diff
@@ -165,17 +165,12 @@ void sendRequestReply(messagebus::MessageBus* msgbus, int argc, char** argv) {
     }
 
     dumpMessage(msg);
-    msgbus->sendRequest(queue, msg);
-
-    setSignalHandler();
-    msgbus->receive(clientName, [](messagebus::Message msg) { 
-        dumpMessage(msg);
-        std::raise(SIGINT);
-    });
-
-    // Wait until interrupt.
-    std::unique_lock<std::mutex> lock(g_mutex);
-    g_cv.wait(lock, [] { return g_exit; });
+    try {
+        dumpMessage(msgbus->request(queue, msg, stoi(timeout)));
```

Comment:
picky note: timeout is string user input... stoi without check is just dangerous to hang messagebus.

Reply:
We assume that the developers which use this tool know what they are doing. Passing a timeout that lasts until the heat deah of the universe is really the least of our concerns when someone has access to the Malamute bus.

Reply:
:) perfect answer :)
Picky note, i mean it's ok... just very minor thing... up to you to fix it.